web3-dev/massa-web3.rst

```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
 
 - `blog <https://github.com/massalabs/massa-sc-examples/tree/main/blog>`_ is an
   example of a decentralized blog platform.
+
+Building Decentralized Application
+##################################
+
+Prerequisites:
+  * having `Thyra <https://github.com/massalabs/thyra/>`_ installed and running on your computer
+  * having a wallet in Thyra
+  * the wallet needs to own some Massa
+  * having a smart-contract :ref:`deployed <sending-sc-sum>`
+
+**Network**: as a blockchain is a network of nodes, there could be several
+networks. The main network is called the mainnet. There is also a testnet,
+run by Massa team, where you can build application without paying real Massa
+tokens. You can also run your own network, with only one node
+(see :ref:`local-network-generation`).
+
+A decentralized application is an application running on a blockchain. Even the
+frontend can be hosted on the blockchain with
+:ref:`web-on-chain <web3-decentralized-web>`.
+
+To build such application you will use some libraries:
+  * `massa toolkit <https://github.com/massalabs/massa-sc-toolkit/>`_
+  * `massa assemblyscript sdk <https://github.com/massalabs/massa-as-sdk/>`_
+  * `massa javascript web3 <https://github.com/massalabs/massa-web3/>`_
+
+Then you will be able to build a frontend application that communicate with
+Thyra's API to sign transactions and submit it to the blockchain. 
+
+Then, use Thyra to deploy your frontend application.
+
+..
+  comment: TODO: add link to the page explaining how to deploy a website
+
+Here you can find an example dapp for the :ref:`sum smart-contract <sc-example-sum>`: 
+`Massa DApp example: Sum <https://github.com/massalabs/massa-dapp-example-sum>`_.
+
+Debugging
+#########
+
+This section will give you some tips to help you debug your dapp while building it.
+
+Work in local
```

Comment:
This sounds strange. Maybe work in local environment?? I am not sure is it a tech phrase 'work in local', better check with @gregLibert

Reply:
You're right I improve the title